angular/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,240 @@
+    });
+  }
+
+  getScale(low: number = 0, high: number, tick: number) {
+    if (this.tick > 0) {
+      let value = high;
+      const ticksArray: number[] = [high];
+      while (value > 0 && (value - tick) >= low) {
+        value -= tick;
+        ticksArray.unshift(Math.abs(Math.round(value / tick) * tick));
+      }
+      this.scale = ticksArray;
+    }
+  }
+
+  getSliderWidth() {
+    return this.sliderBar.nativeElement.getBoundingClientRect().width;
+  }
+
+  getStepWidth() {
+    const maxValue = this.max - this.min;
+    const maxWidth = this.getSliderWidth();
+    return (this.step / maxValue) * maxWidth;
+  }
+
+  getSteps(position: SliderPointerChange) {
+    if (position.isKeyBoard) {
+      return 1;
+    }
+    const diff = position.direction === 1 ? position.to - position.from : position.from - position.to;
+    if (diff < 0) {
+      return 0;
+    }
+    const steps = diff / this.getStepWidth();
+    return steps - Math.floor(steps) >= 0.5 ? Math.ceil(steps) : Math.floor(steps);
+  }
+
+  getLimit(pointerKey: string, direction: number) {
+    if (pointerKey === 'sliderLow') {
+      return this.canCross
+        ? direction === 1 ? this.max : this.min
+        : direction === 1 ? this.sliderHigh : this.min;
+    } else if (pointerKey === 'sliderHigh') {
+      return this.canCross
+        ? direction === 1 ? this.max : this.min
+        : direction === 1 ? this.max : this.sliderLow;
+    }
+  }
+
+  returnCurrentValues() {
+    this.getSelectionWidth();
+    const value = this.isObject(this.value)
+      ?
+      {
+        low: Math.min(this.sliderHigh, this.sliderLow),
+        high: Math.max(this.sliderHigh, this.sliderLow)
+      }
+      :
+      this.sliderHigh;
+    this.onChange(value);
+    this.change.emit(value);
+  }
+
+  moveForward(key: string, pixelMove: number, limit: number) {
+    const newPosition = this[key] + pixelMove <= limit
+      ? this[key] + pixelMove
+      : limit;
+
+    this[key] = newPosition;
+    this.returnCurrentValues();
+  }
+
+  moveBack(key: string, pixelMove: number, limit: number) {
+    const newPosition = this[key] - pixelMove >= limit
+      ? this[key] - pixelMove
+      : limit;
+
+    this[key] = newPosition;
+    this.returnCurrentValues();
+  }
+
+  onSliderMove(key: string, position: SliderPointerChange) {
+    if (this.disabled) {
+      return;
+    }
+    const limit = this.getLimit(key, position.direction);
+    const pixelMove = this.getSteps(position) * this.step;
+
+    position.direction === 1 ? this.moveForward(key, pixelMove, limit) : this.moveBack(key, pixelMove, limit);
+  }
+
+  getSelectionWidth() {
+    const baseValue = Number.isInteger(this.sliderLow) ? this.sliderLow : this.min;
+    this.selectionLeft = `${((Math.min(baseValue, this.sliderHigh) - this.min) / this.max) * 100}%`;
+    this.selectionWidth = `${(Math.abs(this.sliderHigh - baseValue) / this.max) * 100}%`;
+  }
+
+  isObject(value: any): boolean {
```

Comment:
can you create a util file for this function and if you want to use it in template, maybe you can create a pipe using it?

Reply:
I'm using lodash isObject now.